atlp-rwanda/atlp-devpulse-bn
# Extending information which user can update

### PR Description
This PR was about extending information in which user have to update and also allowing him/her to update his profile and view his/her information, because only super Admin was the one to do so

### Description of the task that was expected 

1. Allowing user to update his/profile and View his profile
2. Adding information which user had to update such as email,password,phone,code

### How to test this 

1. Clone repo
2. Checkout branch ft-extend-update-user
3. Run npm install, and then npm run dev to start the server
4. Your open apollo server in browser using localhost:5000. 
5. Ensure you are logged in with your credential  using login mutation
6. To update your profile or view your profile,  you have to pass token you get after  logging in and pass it to header using Authorization as key the token in value field, then use this mutations, one for updating and other for reviewing your information

mutation update_user($id: ID!, $editUserInput: EditUserInput_Logged){
  updateUser_Logged(ID: $id,editUserInput: $editUserInput)
}

query getuser($id: ID!){
user_Logged(ID: $id) {
  id
  firstname
  lastname
  picture
  password
  code
  telephone
  email
  country
  gender
}
}

 Deployed Link: N/A
 
 ###Screenshots
 
![ft-1](https://github.com/user-attachments/assets/1f4262b0-12a7-48a0-9eab-cabe2fb8c0b8)

![updating user](https://github.com/user-attachments/assets/887112b0-fa13-44de-8d33-4d38b8051fb7)

![get user data](https://github.com/user-attachments/assets/e453f875-8631-405b-a296-63e4eda79926)
